File: Packages/Gmf.Net.Core.Common/Requests/PaginatedRequest.cs
using Gmf.DDD.Common.Contracts;

namespace Gmf.Net.Core.Common.Requests;
public class PaginatedRequest : IPaginateRequest
{
    public int PageNumber { get; set; } = 1;
    public int PageSize { get; set; } = 20;
    public string SortBy { get; set; } = string.Empty;
    public bool DescendingSort { get; set; }
}

File: Site/Gmf.Marush.Care.App/src/app/shared/components/pagination/paginated-request.ts
import { signal } from '@angular/core';

export class PaginatedRequest {
    static readonly defaultPageSize = 20;
    readonly filter = signal<string>('');
    readonly pageNumber = signal<number>(1);
    readonly pageSize = signal<number>(PaginatedRequest.defaultPageSize);
    readonly sortBy = signal<string | undefined>(undefined);
    readonly descendingSort = signal<boolean>(false);

    constructor(private readonly initialSort: string) {
        this.sortBy.set(this.initialSort);
    }

    readonly toJson = () => ({
        filter: this.capitalize(this.filter()),
        pageNumber: this.pageNumber(),
        pageSize: this.pageSize(),
        sortBy: this.capitalize(this.sortBy()),
        descendingSort: this.descendingSort()
    });

    private readonly capitalize = (word: string | null | undefined) => {
        if (!word) {
            return '';
        }

        return word.charAt(0).toUpperCase() + word.slice(1);
    };
}
